### PR TITLE
(core) refactor namingService to TS

### PR DESCRIPTION
**Pull request by anotherchrisberry**
**Should not be merged before https://github.com/spinnaker/deck/pull/2960**

If there are conflicts with #2960 I will deal with them after it gets merged...

### CODE REVIEW

File: app/scripts/modules/core/naming/naming.service.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  application: string;
+  stack: string;
+  freeFormDetails: string;
+}
+
+export class NamingService {
+  private VERSION_PATTERN: RegExp = /(v\d{3})/;
+
+  public parseServerGroupName(serverGroupName: string): IComponentName {
+    const result = {
```
Comment (icfantv) [inline app/scripts/modules/core/naming/naming.service.ts]: does this need a type?  i knew for primitive TS types, it's not necessary, didn't realize it for complex types.

File: app/scripts/modules/core/naming/naming.service.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+      result.freeFormDetails = split.slice(2, split.length).join('-');
+    }
+
+    return result;
+  }
+
+  public getClusterName(app: string, stack: string, detail: string): string {
+    let clusterName = app;
+    if (stack) {
+      clusterName += '-' + stack;
```
Comment (icfantv) [inline app/scripts/modules/core/naming/naming.service.ts]: make template string?

File: app/scripts/modules/core/naming/naming.service.ts [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    }
+    if (!stack && detail) {
+      clusterName += '-';
+    }
+    if (detail) {
+      clusterName += '-' + detail;
```
Comment (icfantv) [inline app/scripts/modules/core/naming/naming.service.ts]: ditto?

File: app/scripts/modules/core/naming/naming.service.ts [hunk 4]
```diff
@@ -0,0 +1,90 @@
+import {module} from 'angular';
+
+export const NAMING_SERVICE = 'spinnaker.core.naming.service';
```
Comment (icfantv) [inline app/scripts/modules/core/naming/naming.service.ts]: where do we want to put these?  IMO, it makes sense to put it next to the module definition unless the name is used in the class(es) contained in the file.  thoughts?

File: app/scripts/modules/core/naming/naming.service.spec.ts [hunk 5]
```diff
@@ -1,18 +1,20 @@
 'use strict';
```
Comment (icfantv) [inline app/scripts/modules/core/naming/naming.service.spec.ts]: not needed, tsc adds to transpiled code.

File: app/scripts/modules/core/naming/naming.service.ts (hunk 4 above)
Comment (anotherchrisberry) [inline app/scripts/modules/core/naming/naming.service.ts]: Agreed - will move down.

File: app/scripts/modules/core/naming/naming.service.ts (hunk 1 above)
Comment (anotherchrisberry) [inline app/scripts/modules/core/naming/naming.service.ts]: The TS compiler does not complain about it, but it makes sense to include the type for consistency.

File: app/scripts/modules/core/naming/naming.service.spec.ts (hunk 5 above)
Comment (anotherchrisberry) [inline app/scripts/modules/core/naming/naming.service.spec.ts]: Artifact of file renaming, will remove

File: app/scripts/modules/core/naming/naming.service.ts (hunk 2 above)
Comment (anotherchrisberry) [inline app/scripts/modules/core/naming/naming.service.ts]: sure

**Pull request merged**